tests/unit-tests.tftest.hcl

```
@@ -0,0 +1,608 @@
+            ip_rules                    = ["224.0.0.1/24"]
+            virtual_network_subnet_ids  = null
+        }
+    }
+
+    assert {
+        error_message = "None is a valid value for network_rules.bypass."
+        condition = tolist(azurerm_storage_account_network_rules.tfstate[0].bypass)[0] == "None"
+    }
+}
+
+run "network_rules_bypass_not_allowed" {
+    command = plan
+
+    variables {
+        network_rules = {
+            bypass                      = ["Not Allowed"]
+            default_action              = "Deny"
+            ip_rules                    = ["224.0.0.1/24"]
+            virtual_network_subnet_ids  = null
+        }
+    }
+
+    # Expect the plan to fail
+    expect_failures = [
+        var.network_rules,
+    ]
+}
+
+run "network_rules_ip_rules_invalid_cidr" {
+    command = plan
+
+    variables {
+        network_rules = {
+            bypass                      = null
+            default_action              = "Deny"
+            ip_rules                    = ["224.0.0.1/224"]
+            virtual_network_subnet_ids  = null
+        }
+    }
+
+    # Expect the plan to fail
+    expect_failures = [
+        var.network_rules,
+    ]
+}
+
+run "network_rules_ip_rules_invalid_cidr_prefix" {
+    command = plan
+
+    variables {
+        network_rules = {
+            bypass                      = null
+            default_action              = "Deny"
+            ip_rules                    = ["224.0.0.252/32"]
+            virtual_network_subnet_ids  = null
+        }
+    }
+
+    # Expect the plan to fail
+    expect_failures = [
+        var.network_rules,
+    ]
+}
+
+run "network_rules_ip_rules_invalid_cidr_prefix_31" {
+    command = plan
+
+    variables {
+        network_rules = {
+            bypass                      = null
+            default_action              = "Deny"
+            ip_rules                    = ["224.0.0.248/31"]
+            virtual_network_subnet_ids  = null
+        }
+    }
+
+    # Expect the plan to fail
+    expect_failures = [
+        var.network_rules,
+    ]
+}
+
+run "network_rules_ip_rules_invalid_ipv4" {
+    command = plan
+
+    variables {
+        network_rules = {
+            bypass                      = null
+            default_action              = "Deny"
+            ip_rules                    = ["224.0.0.256/24"]
+            virtual_network_subnet_ids  = null
+        }
+    }
+
+    # Expect the plan to fail
+    expect_failures = [
+        var.network_rules,
+    ]
+}
+
+run "network_rules_multiple_ip_rules" {
+    command = plan
+
+    variables {
+        network_rules = {
+            bypass                      = null
+            default_action              = "Deny"
+            ip_rules                    = ["224.0.0.1/28", "225.0.0.2"]
+            virtual_network_subnet_ids  = ["/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.Storage/storageAccounts/myStorageAccount"]
```

Should this also be null?

```suggestion
            virtual_network_subnet_ids  = null
```

---

basically it's up to personal preference. This particular test checks if we can add multiple ip addresses in ip_rules. What do you preffer ? or why do you want 'null' there ?

---

I just thought I saw a pattern where inputs was kept as minimal/undefined as possible and this one differed from the others. Not important, up to you ;-)